ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-blocks/embeddedActivityBlock/embeddedActivityBlock.component.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component, EventEmitter,
+  Input,
+  OnChanges,
+  OnInit,
+  Output,
+  SimpleChanges } from '@angular/core';
+import { ActivityInstance } from '../../../../models/activityInstance';
+import { ActivityServiceAgent } from '../../../../services/serviceAgents/activityServiceAgent.service';
+import { ActivityForm } from '../../../../models/activity/activityForm';
+import { EMPTY, of } from 'rxjs';
+import { catchError, take } from 'rxjs/operators';
+import { LoggingService } from '../../../../services/logging.service';
+
+@Component({
+  selector: 'ddp-embedded-activity-block',
+  templateUrl: './embeddedActivityBlock.component.html',
+  styleUrls: ['./embeddedActivityBlock.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class EmbeddedActivityBlockComponent implements OnInit, OnChanges {
+  @Input() instance: ActivityInstance;
+  @Input() readonly: boolean;
+  @Input() validationRequested: boolean;
+  @Input() studyGuid: string;
+  @Input() singleSection: boolean;
+  @Output() componentBusy = new EventEmitter<boolean>();
+
+  public activity: ActivityForm;
+  private readonly LOG_SOURCE = 'EmbeddedActivityBlockComponent';
+
+  constructor(private activityServiceAgent: ActivityServiceAgent,
+              private changeDetector: ChangeDetectorRef,
+              private logger: LoggingService) { }
+
+  ngOnInit(): void {
+    this.getActivity();
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.readonly.currentValue && changes.validationRequested.currentValue) { // if we just submitted the form
+      this.componentBusy.emit(true);
+      this.activityServiceAgent.flushForm(this.studyGuid, this.instance.instanceGuid)
```

Comment:
This seems like a very indirect way to submit/flushForm. Still need to dig deeper on my end, but perhaps the button click should trigger this?